collated/main/A0139920A.md

```diff
@@ -0,0 +1,297 @@
+# A0139920A
+###### /java/seedu/todoList/logic/commands/DoneCommand.java
+``` java
+    public DoneCommand(String dataType, int targetIndex) {
+    	this.dataType = dataType;
+        this.targetIndex = targetIndex;
+    }
+
+```
+###### /java/seedu/todoList/logic/commands/DoneCommand.java
+``` java
+    @Override
+    public CommandResult execute() {
+    	
+    	UnmodifiableObservableList<ReadOnlyTask> lastShownList = null;
+    	switch (dataType) {
+    		case "todo":
+    			lastShownList = model.getFilteredTodoList();
+    			break;
+    		case "event":
+    			lastShownList = model.getFilteredEventList();
+    			break;
+    		case "deadline":
+    			lastShownList = model.getFilteredDeadlineList();
+    	}
+        if (lastShownList.size() < targetIndex) {
+            indicateAttemptToExecuteIncorrectCommand();
+            return new CommandResult(Messages.MESSAGE_INVALID_TASK_DISPLAYED_INDEX);
+        }
+
+        ReadOnlyTask taskToDone = lastShownList.get(targetIndex - 1);
+
+        try {
+            model.doneTask(taskToDone, dataType, targetIndex);
+        } catch (TaskNotFoundException pnfe) {
+            assert false : "The target task cannot be missing";
+        }
+
+        return new CommandResult(String.format(MESSAGE_DONE_TASK_SUCCESS, taskToDone));
+    }
+
+}
+```
+###### /java/seedu/todoList/logic/commands/EditCommand.java
+``` java
+    public EditCommand(String name, String date, String endDate, String priority, int targetIndex, String dataType)
+            throws IllegalValueException {
+    	this.targetIndex = targetIndex;
+    	this.dataType = dataType;
+        this.toEdit = new Todo(
+                new Name(name),
+                new StartDate(date),
+                new EndDate(endDate),
+                new Priority(priority),
+                "false"
+        );
+    }
+    
+    /**
+     * Edit Event
+     * Convenience constructor using raw values.
+     * @throws IllegalValueException if any of the raw values are invalid
+     */
+```
+###### /java/seedu/todoList/logic/commands/EditCommand.java
+``` java
+    public EditCommand(String name, String date, String endDate, String startTime, String endTime, int targetIndex, String dataType)
+            throws IllegalValueException {
+    	this.targetIndex = targetIndex;
+    	this.dataType = dataType;
+        this.toEdit = new Event(
+                new Name(name),
+                new StartDate(date),
+                new EndDate(endDate),
+                new StartTime(startTime),
+                new EndTime(endTime),
+                "false"
+        );
+    }
+    
+    /**
+     * Edit Deadline
```

Comment:
Edit_**s**_ ...
please fix all other header comments that do not follw the convention